src/lib/microservices/sync-service/index.ts

```diff
@@ -0,0 +1,235 @@
+    }
+};
+
+const reportGithubTimeoutIssues = async (job: IJob) => {
+    try {
+        /*
+         * In case of timeout, all the rules will have
+         * the same error.
+         */
+        const hint = job.hints[0];
+        const expectedMessage = `webhint didn't return the result fast enough`;
+        const message = hint.messages && hint.messages[0] && hint.messages[0].message;
+
+        if (message && message.includes(expectedMessage)) {
+            const issueReporter: IssueReporter = new IssueReporter();
+            const issueData: IssueData = {
+                configs: job.config,
+                errorMessage: message,
+                errorType: 'timeout',
+                log: job.log,
+                scan: moment().format('YYYY-MM-DD'),
+                url: job.url
+            };
+
+            await issueReporter.report(issueData);
+
+            logger.log('Reported to GitHub successfully', moduleName);
+        }
+    } catch (err) {
+        logger.error('Error reporting to GitHub', moduleName);
+        logger.error(err, moduleName);
+    }
+};
+
+const closeGithubIssues = async (dbJob: IJobModel) => {
+    try {
+        const expectedMessage = `webhint didn't return the result fast enough`;
+
+        // Check first if there was any timeout.
+        const someTimeout = dbJob.hints.some((hint) => {
+            const message = hint.messages && hint.messages[0] && hint.messages[0].message;
+
+            return message && message.includes(expectedMessage);
+        });
+
+        if (!someTimeout) {
+            const issueReporter: IssueReporter = new IssueReporter();
+            const issueData: IssueData = {
+                scan: moment().format('YYYY-MM-DD'),
+                url: dbJob.url
+            };
+
+            await issueReporter.report(issueData);
+
+            logger.log('Issue closed successfully', moduleName);
+        }
+    } catch (err) {
+        logger.error('Error closing issue on GitHub', moduleName);
+        logger.error(err, moduleName);
+    }
+};
+
+const run: AzureFunction = async function(context: Context, job: IJob): Promise<void> {
+    await database.connect(Database);
+
+    const id = job.id;
+    let lock;
+
+    try {
+        lock = await database.lock(id);
+    } catch (e) {
+        logger.error(`It was not possible lock the id ${id}`, moduleName, e);
+        lock = null;
+    }
+
+    if (!lock) {
+        /*
+            * If we are not able to lock the job in the database, keep
+            * the item locked in the queue until the timeout (in the queue)
+            * expire.
+            */
+
+        return;
+    }
+
+    const dbJob: IJobModel = await database.job.get(id);
+
+    if (!dbJob) {
+        logger.error(`Job ${id} not found in database`, moduleName);
+        await database.unlock(lock);
+
+        appInsightClient.trackException({ exception: new Error(`Job ${id} not found in database`) });
+
+        return; // eslint-disable-line no-continue
```

Comment:
The comment is not necessary anymore.

```suggestion
        return;
```